### PR TITLE
Simplify tria.get_communicator() calls according to updates in deal.II

### PR DESCRIPTION
**Pull request by nfehn**
We no longer need casts to get the communicator.

References for instances of MPI_Comm should be removed since this is potentially error prone.

### CODE REVIEW

File: include/exadg/operators/operator_base.cpp [hunk 1]
```diff
@@ -96,20 +96,8 @@ OperatorBase<dim, Number, n_components>::reinit(MatrixFree<dim, Number> const &
...
-
-    if(tria != NULL)
-      comm = tria->get_communicator();
-    else // not distributed triangulation
-      comm = MPI_COMM_SELF;
-  }
-
-  n_mpi_processes = Utilities::MPI::n_mpi_processes(comm);
+  Triangulation<dim> const & tria = dof_handler.get_triangulation();
+  n_mpi_processes                 = Utilities::MPI::n_mpi_processes(tria.get_communicator());
```
Comment (peterrum) [inline include/exadg/operators/operator_base.cpp:100]:
```suggestion
n_mpi_processes                 = Utilities::MPI::n_mpi_processes(dof_handler.get_communicator());
```

File: applications/fluid_structure_interaction/cylinder_with_flag/application_convergence_problems.h [hunk 2]
```diff
@@ -0,0 +1,1045 @@
+/*
+ * cylinder_with_flag.h
+ *
+ *  Created on: Mar 18, 2020
+ *      Author: fehn
```
Comment (kronbichler) [inline applications/fluid_structure_interaction/cylinder_with_flag/application_convergence_problems.h:5]: Is this file added intentionally?

File: include/exadg/operators/operator_base.cpp [hunk 3]
```diff
@@ -708,24 +696,12 @@ template<int dim, typename Number, int n_components>
...
-    else // not distributed triangulation
-      comm = MPI_COMM_SELF;
-  }
+  DoFHandler<dim> const &    dof_handler = this->matrix_free->get_dof_handler(this->data.dof_index);
+  Triangulation<dim> const & tria        = dof_handler.get_triangulation();
 
   TrilinosWrappers::SparsityPattern dsp(is_mg ? dof_handler.locally_owned_mg_dofs(this->level) :
                                                 dof_handler.locally_owned_dofs(),
-                                        comm);
+                                        tria.get_communicator());
```
Comment (peterrum) [inline include/exadg/operators/operator_base.cpp:704]: Same here and below.

File: include/exadg/operators/operator_base.cpp (hunk 1 above)
Comment (kronbichler) [inline include/exadg/operators/operator_base.cpp:100]: Should we simplify this to
```suggestion
  n_mpi_processes                 = Utilities::MPI::n_mpi_processes(dof_handler.get_triangulation().get_communicator());
```
or, even simpler,
```suggestion
  n_mpi_processes                 = Utilities::MPI::n_mpi_processes(dof_handler.get_communicator());
```

File: include/exadg/operators/operator_base.cpp (hunk 3 above)
Comment (kronbichler) [inline include/exadg/operators/operator_base.cpp:704]: If `tria` is not used elsewhere in this function, I suggest to immediately use the `dof_handler.get_communicator()` function.

File: applications/fluid_structure_interaction/cylinder_with_flag/application_convergence_problems.h (hunk 2 above)
Comment (nfehn) [reply applications/fluid_structure_interaction/cylinder_with_flag/application_convergence_problems.h]: no, has been a mistake

File: include/exadg/operators/operator_base.cpp (hunk 1 above)
Comment (kronbichler) [reply include/exadg/operators/operator_base.cpp:100]: At least we're consistent in the comments, @peterrum :wink:

**Comment by nfehn** [comment]
please reconsider, I think it does not make sense to answer point-by-point

**Review by kronbichler — APPROVED** [review]

**Pull request merged**